documentation/writing_tasks.md

```diff
@@ -1077,3 +1077,453 @@ path.
+      self.debug("Quotient of values: {}".format(quotient))
+
+      return { 'sum': sum, 'quotient': quotient }
+
+if __name__ == '__main__':
+    MyTask().run()
+
+```
+
+Running this task with the parameter `value_2=0` will raise a `TaskError`
+that will automatically include the logged debugging statements under the
+`details` key:
+
+```shell
+$ bolt task run mytask -t localhost value_1=10 value_2=0 --format json
+{
+  "items":[
+    {
+      "target":"localhost",
+      "action":"task",
+      "object":"mytask",
+      "status":"failure",
+      "value":{
+        "_error":{
+          "msg":"integer division or modulo by zero",
+          "issue_code":"EXCEPTION",
+          "kind":"python.task.helper/exception",
+          "details":{
+            "debug":[
+              "Adding values",
+              "Sum of values: 1",
+              "Dividing values"
+            ],
+            "class":"ZeroDivisionError"
+          }
+        }
+      }
+    }
+  ],
+  "target_count":1,
+  "elapsed_time":0
+}
+```
+
+📖 **Related information**
+- [Python task helper
+  debugging](https://github.com/puppetlabs/puppetlabs-python_task_helper#debugging)
+
+#### Ruby tasks
+
+The following Ruby task includes a few debugging statements which describe
+what the task is doing and the results from a couple arithmetic operations:
+
+```ruby
+#!/usr/bin/env ruby
+
+require_relative "../../ruby_task_helper/files/task_helper.rb"
+
+class MyTask < TaskHelper
+  def task(opts)
+    debug 'Adding values'
+    sum = opts[:value_1] + opts[:value_2]
+    debug "Sum of values: #{sum}"
+
+    debug 'Dividing values'
+    quotient = opts[:value_1] / opts[:value_2]
+    debug "Quotient of values: #{quotient}"
+
+    { sum: sum, quotient: quotient }
+  end
+end
+
+if __FILE__ == $0
+  MyTask.run
+end
+```
+
+Running this task with the parameter `value_2=0` will raise a `TaskError`
+that will automatically include the logged debugging statements under the
+`details` key:
+
+```shell
+$ bolt task run mytask -t localhost value_1=10 value_2=0 --format json
+
+{
+  "items":[
+    {
+      "target":"localhost",
+      "action":"task",
+      "object":"mytask",
+      "status":"failure",
+      "value":{
+        "_error":{
+          "kind":"ZeroDivisionError",
+          "msg":"divided by 0",
+          "details":{
+            "debug":[
+              "Adding values",
+              "Sum of values: 1",
+              "Dividing values"
+            ]
+          }
+        }
+      }
+    }
+  ],
+  "target_count":1,
+  "elapsed_time":0
+}
+```       
+
+📖 **Related information**
+- [Ruby task helper
+  debugging](https://github.com/puppetlabs/puppetlabs-ruby_task_helper#debugging)
+
+### Running a task as a script
+
+Running tasks without Bolt can make debugging easier, as the task will no
+longer be executed in a separate thread. Since tasks are similar to scripts,
+you can make temporary changes to the task so they can accept command line
```

Comment:
```suggestion
you can make temporary changes to the task so they can accept command-line
```
This is a style thing - "command-line" for adjectives, "command line" for nouns. Our docs aren't super consistent on this. I'll open a pothole PR for this at some point.